src/frame/reason.rs

```diff
@@ -1,11 +1,21 @@
 use std::fmt;
 
 
-/// HTTP2 Error codes
+/// HTTP/2.0 error codes.
+///
+/// Error codes are used in `RST_STREAM` and `GOAWAY` frames to convey the
+/// reasons for the stream or connection error. For example,
+/// [`SendStream::send_reset`] takes a `Reason` argument. Also, the `Error` type
+/// may contain a `Reason`.
+///
+/// Error codes share a common code space. Some error codes apply only to either
+/// streams or the entire connection and have no defined semantics in the other
+/// context.
```

Comment:
I get what you're trying to say but this sentence is a little confusing.

Maybe clearer:

> Error codes share a common code space. Some error codes apply only to streams, others apply only to connections, and others may apply to either.  See [RFC 7540] for more information.